core/src/main/scala/akka/persistence/r2dbc/internal/postgres/PostgresQueryDao.scala

```diff
@@ -430,28 +412,38 @@ private[r2dbc] class PostgresQueryDao(settings: R2dbcSettings, executorProvider:
   }
 
   override def persistenceIds(entityType: String, afterId: Option[String], limit: Long): Source[String, NotUsed] = {
+    val actualLimit = if (limit > Int.MaxValue) Int.MaxValue else limit.toInt
     val likeStmtPostfix = PersistenceId.DefaultSeparator + "%"
-    // FIXME
-    require(settings.numberOfDataPartitions == 1, "persistenceIds not implemented for more than one data-partition yet")
-    val executor = executorProvider.executorFor(slice = 0)
-    val result = executor.select(s"select persistenceIds by entity type")(
-      connection =>
-        afterId match {
-          case Some(after) =>
-            val stmt = connection.createStatement(persistenceIdsForEntityTypeAfterSql)
-            bindPersistenceIdsForEntityTypeAfterSql(stmt, entityType, likeStmtPostfix, after, limit)
-
-          case None =>
-            val stmt = connection.createStatement(persistenceIdsForEntityTypeSql)
-            bindPersistenceIdsForEntityTypeSql(stmt, entityType, likeStmtPostfix, limit)
+    val results: immutable.IndexedSeq[Future[immutable.IndexedSeq[String]]] =
+      // query each data partition
+      settings.dataPartitionSliceRanges.map { sliceRange =>
+        val executor = executorProvider.executorFor(sliceRange.min)
+        executor.select(s"select persistenceIds by entity type")(
+          connection =>
+            afterId match {
+              case Some(after) =>
+                val stmt = connection.createStatement(persistenceIdsForEntityTypeAfterSql(sliceRange.min))
+                bindPersistenceIdsForEntityTypeAfterSql(stmt, entityType, likeStmtPostfix, after, actualLimit)
+
+              case None =>
+                val stmt = connection.createStatement(persistenceIdsForEntityTypeSql(sliceRange.min))
+                bindPersistenceIdsForEntityTypeSql(stmt, entityType, likeStmtPostfix, actualLimit)
+
+            },
+          row => row.get("persistence_id", classOf[String]))
+      }
```

Comment:
So limit will now in fact potentially yield `db partion * table partition * limit` entries, worth mentioning somewhere?

Reply:
It's table partitions * limit. data partion = table, then those tables can optionally be spread over multiple databases, but there will not be a multipler

Yeah, I know it's not optimal to query all partitions for the same thing and then sort and truncate the result, but as far as I can see it that's the only way to do it. Ordering and paging as mentioned in other comment.

Reply:
and these persistenceIds queries are intended for backoffice, not heavy use

Reply:
Ok, combined with the fact that the partitioning is very specific performance trixery, maybe it is fine.